File: app/src/main/java/com/ludoscity/findmybikes/ui/settings/SettingsActivity.java
package com.ludoscity.findmybikes.ui.settings;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NavUtils;

import com.ludoscity.findmybikes.R;

/**
 * Created by F8Full on 2015-08-10.
 * Activity used to display Settings fragment
 */
public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_settings);

        setSupportActionBar(findViewById(R.id.toolbar_main));

        //noinspection ConstantConditions
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);

        //noinspection ConstantConditions
        getSupportActionBar().setSubtitle(getString(R.string.hashtag_settings));
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Respond to the action bar's Up/Home button
        if (item.getItemId() == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/ludoscity/findmybikes/ui/settings/SettingsActivity.kt
package com.ludoscity.findmybikes.ui.settings

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.NavUtils
import androidx.lifecycle.ViewModelProviders
import com.ludoscity.findmybikes.R
import com.ludoscity.findmybikes.utils.InjectorUtils

/**
 * Created by F8Full on 2015-08-10.
 * Activity used to display Settings fragment
 */
class SettingsActivity : AppCompatActivity() {

    //capturing intent targeting custom URL scheme defined in manifest
    //for OAuth login flow
    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)

        val action = intent?.action
        val data = intent?.dataString

        if (action == Intent.ACTION_VIEW && data != null) {
            //Log.e("SettingsActivity", "Intent data string : $data")

            ViewModelProviders.of(this, InjectorUtils.provideSettingsActivityViewModelFactory(application))
                    .get(SettingsActivityViewModel::class.java).retrieveAccessTokenAndRefreshToken(data)
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_settings)

        setSupportActionBar(findViewById(R.id.toolbar_main))


        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)


        supportActionBar!!.subtitle = getString(R.string.hashtag_settings)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Respond to the action bar's Up/Home button
        if (item.itemId == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this)
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
